code/controllers/subsystem/dbcore.dm

```diff
@@ -135,33 +106,22 @@ SUBSYSTEM_DEF(dbcore)
 	qdel(query_round_last_id)
 
 /datum/controller/subsystem/dbcore/proc/SetRoundStart()
-	if(!Connect())
+	if(!rustg_sql_connected())
 		return
 	var/datum/DBQuery/query_round_start = SSdbcore.NewQuery("UPDATE [format_table_name("round")] SET start_datetime = Now() WHERE id = [GLOB.round_id]")
 	query_round_start.Execute()
 	qdel(query_round_start)
 
 /datum/controller/subsystem/dbcore/proc/SetRoundEnd()
-	if(!Connect())
+	if(!rustg_sql_connected())
 		return
 	var/sql_station_name = sanitizeSQL(station_name())
 	var/datum/DBQuery/query_round_end = SSdbcore.NewQuery("UPDATE [format_table_name("round")] SET end_datetime = Now(), game_mode_result = '[sanitizeSQL(SSticker.mode_result)]', station_name = '[sql_station_name]' WHERE id = [GLOB.round_id]")
 	query_round_end.Execute()
 	qdel(query_round_end)
 
 /datum/controller/subsystem/dbcore/proc/Disconnect()
-	failed_connections = 0
-	QDEL_NULL(connectOperation)
-	QDEL_NULL(connection)
-
-/datum/controller/subsystem/dbcore/proc/IsConnected()
-	if(!CONFIG_GET(flag/sql_enabled))
-		return FALSE
-	//block until any connect operations finish
-	var/datum/BSQL_Connection/_connection = connection
-	var/datum/BSQL_Operation/op = connectOperation
-	UNTIL(QDELETED(_connection) || op.IsComplete())
-	return !QDELETED(connection) && !op.GetError()
+	rustg_sql_disconnect_pool()
 

```

Comment:
```suggestion
/datum/controller/subsystem/dbcore/proc/IsConnected()
	if(!CONFIG_GET(flag/sql_enabled))
		return FALSE
	var/result = json_decode(rustg_sql_connected())
	world.log << result
	return result["status"] == "online"


```

Reply:
please surface the errors to the sql log goof